maths/perfect_square.py

```diff
@@ -21,6 +21,36 @@ def perfect_square(num: int) -> bool:
     return math.sqrt(num) * math.sqrt(num) == num
 
 
+def perfect_square_binary_search(n: int) -> bool:
+    """
+    Check if a number is perfect square using binary search.
+    Time complexity : O(Log(n))
+    Space complexity: O(1)
+
+    >>> perfect_square_binary_search(9)
+    True
+    >>> perfect_square_binary_search(16)
+    True
+    >>> perfect_square_binary_search(1)
+    True
+    >>> perfect_square_binary_search(0)
+    True
+    >>> perfect_square_binary_search(10)
+    False
```

Comment:
tests for -1, 1.1, "a", None, []?

Reply:
should I first add input validation inside the function, or should I just include those tests ?

Reply:
Write the test first and see how your function deals with unexpected input.  The trick is to not write validation code if you don't have to. If your function is already going to raise an IndexError, TypeError, ValueError, etc. then you have no work to do. Just look for cases where your code would silently fail or would deliver nonsense results.

Reply:
Got it.Thank you so much ! 
Things should be good now.